scripts/dashboard/utils.py

```diff
@@ -0,0 +1,30 @@
+import re
+import os
+
+from medperf import config
+
+
+def stage_id2name(stage_str, stages_df):
+    _, code = stage_str.split()
+    code = float(code)
```

Comment:
Not sure if there's going to be an easy way out of this, and it comes from a failure of thinking about this when designing the stages. Stages code can be an int or a float as defined in the stages.csv file, while reports we get back from the server can also be int or float depending on the stages encountered by the user. That means that for example we have stage `'8'` but the report says it is `stage 8.0` in which case string matching will not work, or would require at some point float casting to harmonize both strings, which would introduce the aforementioned bugs.

It actually appears that pandas can handle float indices for the most part, behaving as expected in most situations. https://pandas.pydata.org/pandas-docs/version/0.13.1/indexing.html#float64index

While I understand the issue with floats, and I'm dissapointed I didn't consider this when I started designing the whole pipeline, given the scenario we're dealing with here, with well defined and predefined floats instead of floats coming out of computations, I think we can work with floating indices.

Do let me know if you have a thought on how to harmonize the strings from the stages and the report so that we can avoid floating points altogether. We could make changes to how reports are generated, but we would still need to have backwards compatibility with those reports that were generated before the change.

Reply:
And we cannot fix reports easily, right? I mean, what would happen if starting from tomorrow report would send stage code as str instead of float?

I understand such a fix is out of scope for this PR, so I'm asking mostly for getting if we can improve this in future